### PR TITLE
fix: RDCC-2517 fix child deletion when update record fails in UP

### PR DESCRIPTION
**Pull request by shreedhar-hmcts**
### JIRA link (if applicable) ###
RDCC-2517

### Change description ###
fix: RDCC-2517 fix child deletion when update record fails in UP

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by SujeethKumarAri — APPROVED** [review]

**Review by nitishsingla1234 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java [hunk 1]
```diff
@@ -411,15 +430,14 @@ public void populateCaseWorkerProfile(CaseWorkersProfileCreationRequest cwrdProf
         return caseWorkerWorkAreas;
     }
 
-    public boolean updateUserProfile(CaseWorkersProfileCreationRequest cwrdProfileRequest,
+    public CaseWorkerProfile updateUserProfile(CaseWorkersProfileCreationRequest cwrdProfileRequest,
                                      CaseWorkerProfile caseWorkerProfile) {
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java:434]: Refactor method name it is very misleading and hard to maintain

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java [hunk 2]
```diff
@@ -200,28 +200,47 @@
...
-        if (isNotEmpty(updateCaseWorkerProfiles)) {
-            caseWorkerLocationRepository.deleteByCaseWorkerProfileIn(updateCaseWorkerProfiles);
-            caseWorkerWorkAreaRepository.deleteByCaseWorkerProfileIn(updateCaseWorkerProfiles);
-            caseWorkerRoleRepository.deleteByCaseWorkerProfileIn(updateCaseWorkerProfiles);
+        // update roles in sidam and filter if failed in User profile
+        List<CaseWorkerProfile> filteredUpdateCwProfiles = updateSidamRoles(updateCaseWorkerProfiles, requestMap);
+
+        if (isNotEmpty(filteredUpdateCwProfiles)) {
+            caseWorkerLocationRepository.deleteByCaseWorkerProfileIn(filteredUpdateCwProfiles);
+            caseWorkerWorkAreaRepository.deleteByCaseWorkerProfileIn(filteredUpdateCwProfiles);
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java:208]: Refactor all child deletes in single method and add some comments why we are doing that will be useful in future to maintain and also please mention right point of delete is after sidam updates.

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java (hunk 1 above)
Comment (shreedhar-hmcts) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java:439]: yes , anyways this code will be refactored in review comments change.

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java (hunk 2 above)
Comment (shreedhar-hmcts) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java:208]: yes , anyways this code will be refactored in review comments change.

**Review by nitishsingla1234 — APPROVED** [review]

**Review by SujeethKumarAri — APPROVED** [review]

**Pull request merged**